src/main/InstanceService.ts

```diff
@@ -258,23 +259,53 @@ class InstanceService {
     const instance = this.getInstance(instanceId);
 
     if (instance.account.service) {
-      return instance.client.client.api.getClients();
+      return instance.account.service.client.getClients();
     } else {
       throw new Error(`Account service for instance ${instanceId} not set.`);
     }
   }
 
-  async removeAllOtherClients(instanceId: string, password: string): Promise<void> {
-    const instance = this.getInstance(instanceId);
+  async removeAllClients(backend: string, email: string, password: string): Promise<void> {
+    const backendType = backend === 'staging' ? APIClient.BACKEND.STAGING : APIClient.BACKEND.PRODUCTION;
+    const engine = new MemoryEngine();
+    await engine.init('temporary');
+    const apiClient = new APIClient({store: engine, urls: backendType});
+    const account = new Account(apiClient);
 
-    if (instance.account.service) {
-      const clients = await instance.client.client.api.getClients();
-      for (const client of clients) {
-        await instance.client.client.api.deleteClient(client.id, password);
+    const ClientInfo: ClientInfo = {
+      classification: ClientClassification.DESKTOP,
+      cookieLabel: 'default',
+      model: `E2E Test Server v${version}`,
+    };
+
+    const loginData = {
+      clientType: ClientType.PERMANENT,
+      email,
+      password,
+    };
+
+    try {
+      await account.login(loginData, true, ClientInfo);
+    } catch (error) {
+      if (error.code !== StatusCode.FORBIDDEN || error.label !== BackendErrorLabel.TOO_MANY_CLIENTS) {
+        throw error;
       }
-    } else {
-      throw new Error(`Account service for instance ${instanceId} not set.`);
     }
+
+    const clients = await apiClient.client.api.getClients();
+    const instances = this.cachedInstances.getAll();
+
+    for (const client of clients) {
+      for (const instanceId in instances) {
+        const instance = this.cachedInstances.get(instanceId);
+        if (instance && instance.client.context && instance.client.context.clientId === client.id) {
+          await this.deleteInstance(instanceId);
+        }
+      }
+      await apiClient.client.api.deleteClient(client.id, password);
+    }
+
+    await account.logout();
```

Comment:
is logout possible when the login error for "too many clients" happens and is cought?

Reply:
Yes because it's just a `POST` request to the backend and deleting access tokens and cookies.

Reply:
👍